### DISCUSSION TITLE
Dannjs Q&A

### DISCUSSION THREAD

**Discussion by matiasvlevi** [matiasvlevi/Dann#3] [Q&A]
# Dannjs Q&A

**Comment by lortonx** [comment]
my question is too broad, I would like many examples

**Reply by matiasvlevi** [reply]
What is your question then?

**Reply by sepatin** [reply]
Hey , 

Thanks for the library and nice examples 
Just a question ... i did not found a way to manage quantile loss with the library... Is it implemented ?
Thanks

**Reply by matiasvlevi** [reply]
Hey, 
Quantile loss was not implemented in the library. After seeing your comment, I added it right away. As of `v2.2.11` quantile loss is implemented. Let me know if there are modifications or adjustments to make about this new feature.

You can read about the new Quantile loss function [here](https://dannjs.org/docs/classes/Dann#method_setLossFunction). The second example explains how to set a percentile value.

**Comment by davay42** [comment]
Hey! I've seen a couple of neural net video explainers on youtube and have some JS coding experience so DANN seems to be a nice next step to try it. I understand that the docs are straightforward for any experienced neural net programmer. But for me to flll the gap some starter project is very needed. Like the number recognition or some other trivial example of training and using a net.  Is there any `codepen` or any other live code to experiment with?

**Reply by matiasvlevi** [reply]
Hey, I just added a [JSBin ](https://jsbin.com/vuzibunobo/edit?js,console) interactive example, jsbin it is pretty similar to codepen. I have some more examples on my website [here](https://dannjs.org#demos).
Ill be making more interactive examples for different situations (mnist is coming), they'll appear on the website.

**Reply by davay42** [reply]
Yeah! That's a very nice and simple starter to experiment with! Thanks!

**Comment by dhruvroytalukdar** [comment]
Please can you explain what is this NN.makeWeights() function doing?

**Reply by matiasvlevi** [reply]
`nn.makeWeights();` will initialize/create the weights & biases. 

Most of the time if you haven't used this method and use an other method that requires the weights, its going to create them automatically and leave a warning in the console.

Documentation about the method can be found [here](https://dannjs.org/docs/classes/Dann.html#method_makeWeights).

**Reply by dhruvroytalukdar** [reply]
Yeah. I found that I got correct results even without using this method (in the code present on your site). So I was a bit confused about what is this function doing. Thanks by the way.

**Comment by solid-droid** [comment]
Hey, can the library do batch back-propagation/training ?
Instead of sending one input at a time , by sending a batch of inputs , makes the neural network train faster.

**Reply by matiasvlevi** [reply]
We could definitely add a method for that.
I think having a backpropagation method for 1 data sample is essential for simplicity and understanding, but having a method to train batches would be useful too.
Do you want to submit a "Feature request" issue about this method?

**Reply by solid-droid** [reply]
yup, it will be really useful. -> https://github.com/matiasvlevi/Dann/issues/41

**Comment by andrew-medvedev** [comment]
Do you plan to include q-learning? (tensorflor.js still doesn't have one)

**Reply by matiasvlevi** [reply]
Hey Andrew,
Thanks for the question!

I personnally don't have any experience with Q-learning and Deep Q-learning. I have briefly read about it, and it sounds interesting. It would make for a nice feature. Do you want to post an issue about this new feature?

Also if anyone has any experience with these algorithms and wants to see them in Dannjs, feel free to contribute!
